tests/tools/onert_train/src/metric.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __ONERT_TRAIN_METRIC_H__
+#define __ONERT_TRAIN_METRIC_H__
+
+#include "allocation.h"
+#include "nnfw.h"
+#include <vector>
+
+namespace onert_train
+{
+class Metric
+{
+public:
+  Metric(const std::vector<Allocation> &output, const std::vector<Allocation> &expected,
+         const std::vector<nnfw_tensorinfo> &infos);
+
+private:
+  template <typename T>
+  float categoricalAccuracy(const T *output, const T *expected, uint32_t batch, uint64_t size);
+
+public:
+  float categoricalAccuracy(int32_t index);
+
+private:
+  const std::vector<Allocation> &_output;
+  const std::vector<Allocation> &_expected;
```

Comment:
@jyoungyun Is it necessary to have `_output` and `_expected` as members? `categoricalAccuracy(const T *output, const T *expected, uint32_t batch, uint64_t size)` seems to have everything needs to calculate. I expect `Metric` is set of algorithms for calculating several kind of accuracies.

Reply:
Yes, the metric class contains the _output and _expected results as members to access each index data easily.
Here is an example for this class. I want to match the usability of this function with `nnfw_train_get_loss` function.

```
            // get validation loss and accuracy
            Metric metric(output_data, expected_data, expected_infos);
            for (int32_t i = 0; i < num_expecteds; ++i)
            {
              float temp = 0.f;
              NNPR_ENSURE_STATUS(nnfw_train_get_loss(session, i, &temp));
              losses[i] += temp;
              if (args.getMetricType() == 0)
                metrics[i] += metric.categoricalAccuracy(i);
            }
          }
```